src/central_freelist.cc

```diff
@@ -345,19 +345,22 @@ void CentralFreeList::Populate() {
 
   // Split the block into pieces and add to the free-list
   // TODO: coloring of objects to avoid cache conflicts?
-  void** tail = &span->objects;
   char* ptr = reinterpret_cast<char*>(span->start << kPageShift);
   char* limit = ptr + (npages << kPageShift);
   const size_t size = Static::sizemap()->ByteSizeForClass(size_class_);
-  int num = 0;
+  int num = 1;
+  // Special case for the first element
+  span->objects = ptr;
+  void* tail = ptr;
+  ptr += size;
```

Comment:
unfortunately this fragment doesn't apply cleanly anymore and for a reason. There was possibility of overflow in ptr + size computation which is now fixed. So your code will need to adapt since you're special casing first item and thus duplicating ptr + size logic (feel free keep duplication)